Sources/Testing/Traits/CoverageTrait.swift

```diff
@@ -0,0 +1,241 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+internal import _TestingInternals
+
+// MARK: - Coverage Detection
+
+/// Check if coverage instrumentation is enabled for this test run.
+///
+/// This is determined by checking for the presence of coverage-related
+/// environment variables.
+private let _coverageEnabled: Bool = {
+    // Check for LLVM_PROFILE_FILE environment variable (set by swift test)
+    Environment.variable(named: "LLVM_PROFILE_FILE") != nil
```

Comment:
Please make sure to document any new environment variable usage in EnvironmentVariables.md.